engine/lib/comments.php

```diff
@@ -290,6 +292,63 @@ function _elgg_comments_notification_email_subject($hook, $type, $returnvalue, $
 	return $returnvalue;
 }
 
+/**
+ * Update comment access to match that of the container
+ * 
+ * @param string     $event  'update:after'
+ * @param string     $type   'all'
+ * @param ElggEntity $entity The updated entity
+ * @return bool
+ * 
+ * @access private
+ */
+function _elgg_comments_access_sync($event, $type, $entity) {
+	if ($entity instanceof \ElggEntity) {
+		
+		// need to override access in case comments ended up with ACCESS_PRIVATE
+		// and to ensure write permissions
+		$ia = elgg_set_ignore_access(true);
+		$options = array(
+			'type' => 'object',
+			'subtype' => 'comment',
+			'container_guid' => $entity->getGUID(),
+			'wheres' => array(
+				"e.access_id != {$entity->access_id}"
+			),
+			'limit' => 0,
```

Comment:
I looked at what the discussions batch for this did and it used 0, but I do prefer ```false```